lib/ansible/module_utils/serialization/fallback_to_str.py

```diff
@@ -0,0 +1,75 @@
+"""
+Lossy best-effort serialization for Ansible variables; used primarily for callback JSON display.
+Any type which is not supported by JSON will be converted to a string.
+The string representation of any type that is not native to JSON is subject to change and should not be considered stable.
+The decoder provides no special behavior.
+"""
+
+from __future__ import annotations as _annotations
+
+import collections.abc as _c
+import datetime as _datetime
+import typing as _t
+
+from json import dumps as _dumps
+
+from ansible.module_utils._internal import _datatag
+from ansible.module_utils.common import json as _json
+
+
+class _Profile(_json._JSONSerializationProfile["Encoder", "Decoder"]):
+    @classmethod
+    def post_init(cls) -> None:
+        cls.serialize_map: dict[type, _t.Callable] = {
```

Comment:
I wonder if this'd work
```suggestion
        cls.serialize_map: dict[type[object], _t.Callable] = {
```